synapse/lib/stormlib/imap.py

```diff
@@ -1,30 +1,441 @@
+import random
 import asyncio
+import imaplib
+import logging
 
-import aioimaplib
+import regex
 
 import synapse.exc as s_exc
 import synapse.common as s_common
 import synapse.lib.coro as s_coro
+import synapse.lib.link as s_link
 import synapse.lib.stormtypes as s_stormtypes
 
-async def run_imap_coro(coro):
+logger = logging.getLogger(__name__)
+
+CRLF = b'\r\n'
+CRLFLEN = len(CRLF)
+UNTAGGED = '*'
+
+def quote(text, escape=True):
+    if text == '""':
+        # Don't quote empty string
+        return text
+
+    if ' ' not in text and '"' not in text:
+        return text
+
+    if escape:
+        text = text.replace('"', '\\"')
+
+    return f'"{text}"'
+
+def qsplit(text):
+    '''
+    Split text on whitespace preserving quoted substrings and escaped quotes. Quoted strings are
+    unquoted and escaped quotes are unescaped.
+    '''
+    ret = []
+
+    token = []
+    inquotes = False
+
+    for idx, char in enumerate(text):
+        # Found a space and we're not in a quoted token
+        if char == ' ' and not inquotes:
+            ret.append(''.join(token))
```

Comment:
should this check that token isn't empty or should multiple spaces in a row add an empty string?

Reply:
Updated to ignore multiple spaces when not in quotes.